utilities/blob_db/blob_db_impl.h

```diff
@@ -408,6 +411,17 @@ class BlobDBImpl : public BlobDB {
 
   std::list<std::shared_ptr<BlobFile>> obsolete_files_;
 
+  // DeleteObsoleteFiles, DiableFileDeletions and EnableFileDeletions block
+  // on the mutex to avoid contention.
+  mutable port::Mutex delete_file_mutex_;
```

Comment:
Document the lock holding order between `delete_file_mutex_` and `mutex_`.

Reply:
No particular order between `delete_file_mutex_` and `mutex_` but they are guarding different stuff. `delete_file_mutex_` simply avoid `DeleteObsolteFiles` and `DisableDeleteFiles` run simultaneously, and `mutex_` guard the data structure. I'll add a comment to explain.